src/components/profile/profile-alert/alert-dialog.tsx

```diff
@@ -0,0 +1,47 @@
+import React from "react";
+import { useI18nStore } from "../../../i18n/i18n-store.ts";
+import { PrimaryButton } from "../../buttons/primary.tsx";
+import { useUrlState } from "../../router/store";
+
+export interface AlertDialogProps {
+	alertTitleWithIcon: string | React.ReactNode;
+	alertMessage: string;
+	href?: string;
+}
+
+export const AlertDialog: React.FC<AlertDialogProps> = ({
+	alertTitleWithIcon,
+	alertMessage,
+	href,
+}) => {
+	const { setPathname } = useUrlState();
+
+	return (
+		<dialog
+			id={"alert-dialog"}
+			className="py-6 px-10 bg-gdk-lighter-gray shadow-gdk-hard rounded-2xl w-11/12 sm:w-[400px] backdrop:backdrop-blur"
+		>
+			<div className="flex-col flex">
+				<span className="font-semibold text-xl flex flex-row gap-x-5 my-4 justify-start">
+					{alertTitleWithIcon}
+				</span>
+				<p className="">{alertMessage}</p>
+
+				<div className="self-end my-2 lg:m-0">
+					<PrimaryButton
+						label={
+							useI18nStore.getState().i18n().navbar.profile.settings.confirm
+						}
+						onClick={() => {
+							(
+								document.getElementById("alert-dialog") as HTMLDialogElement
+							).close();
+
+							href && setPathname(href);
```

Comment:
this line made me wonder for a sec, i think this notation is rather used to conditionally inline react components in JSX. As in this onClick scope we're writing javascript/typescript, I'd suggest to rather use 
```javascript
if (href) {
  setPathname(href)
}
```

and also as the onclick starts to spread over multiple lines, i'd suggest to rather define it before the return instead of here inline.